### PR TITLE
some-logging-improvements

### PR DESCRIPTION
**Pull request by tabmatfournier**
- log during table creation failures
- some additional logging at critical states between coordinator/worker during commit phase

### CODE REVIEW

File: kafka-connect/src/main/java/io/tabular/iceberg/connect/channel/Coordinator.java [hunk 1]
```diff
@@ -127,6 +127,7 @@ private boolean receive(Envelope envelope) {
 
   private void commit(boolean partialCommit) {
     try {
+      LOG.info("Processing commit after responses for {}, isPartialCommit {}",commitState.currentCommitId(), partialCommit);
```
Comment (fqtab) [inline kafka-connect/src/main/java/io/tabular/iceberg/connect/channel/Coordinator.java:130]: nit: I feel this one is unnecessary given that we the commit message for `Commit {} ready, received responses for all {} partitions`.

File: kafka-connect/src/main/java/io/tabular/iceberg/connect/channel/Worker.java [hunk 2]
```diff
@@ -134,7 +134,7 @@ private void routeRecordStatically(SinkRecord record) {
 
   private void routeRecordDynamically(SinkRecord record) {
     String routeField = config.tablesRouteField();
-    Preconditions.checkNotNull(routeField, "Route field cannot be null with dynamic routing");
+    Preconditions.checkNotNull(routeField, String.format("Route field cannot be null with dynamic routing at topic: %s, partition: %d, offset: %d", record.topic(), record.kafkaPartition(), record.kafkaOffset()));
```
Comment (fqtab) [inline kafka-connect/src/main/java/io/tabular/iceberg/connect/channel/Worker.java:137]: 👍

File: kafka-connect/src/main/java/io/tabular/iceberg/connect/data/IcebergWriterFactory.java [hunk 3]
```diff
@@ -67,47 +68,52 @@ public RecordWriter createWriter(
...
     } catch (Exception e) {
-      LOG.error(
-          "Unable to create partition spec {}, table {} will be unpartitioned",
-          partitionBy,
-          identifier,
-          e);
-      spec = PartitionSpec.unpartitioned();
+      LOG.error("Error creating new table {} from record at topic: {}, partition: {}, offset: {}", tableName, sample.topic(), sample.kafkaPartition(), sample.kafkaOffset());
+      throw e;
     }
```
Comment (fqtab) [inline kafka-connect/src/main/java/io/tabular/iceberg/connect/data/IcebergWriterFactory.java:117]: 👍

**Review by fqtab — APPROVED** [review]

File: kafka-connect/src/main/java/io/tabular/iceberg/connect/channel/CommitState.java [hunk 4]
```diff
@@ -125,14 +126,14 @@ public boolean isCommitReady(int expectedPartitionCount) {
             .sum();
 
     if (receivedPartitionCount >= expectedPartitionCount) {
-      LOG.debug(
+      LOG.info(
```
[the remark this answers is not in the document]
Comment (tabmatfournier) [reply kafka-connect/src/main/java/io/tabular/iceberg/connect/channel/CommitState.java:129]: I know.  I think I went too far there.  I'm still keeping SOME of the debugs.

File: kafka-connect/src/main/java/io/tabular/iceberg/connect/data/IcebergWriterFactory.java [hunk 5]
```diff
@@ -56,7 +56,8 @@ public RecordWriter createWriter(
       if (config.autoCreateEnabled()) {
         table = autoCreateTable(tableName, sample);
       } else if (ignoreMissingTable) {
-        return new RecordWriter() {};
+        return new RecordWriter() {
+        };
```
[the remark this answers is not in the document]
Comment (tabmatfournier) [reply kafka-connect/src/main/java/io/tabular/iceberg/connect/data/IcebergWriterFactory.java:60]: weird, formatter doing this too me.

File: kafka-connect/src/main/java/io/tabular/iceberg/connect/channel/Coordinator.java [hunk 6]
```diff
@@ -97,10 +97,10 @@ public void process() {
     if (commitState.isCommitIntervalReached()) {
       // send out begin commit
       commitState.startNewCommit();
+      LOG.info("Started new commit with commit-id={}", commitState.currentCommitId().toString());
```
[the remark this answers is not in the document]
Comment (tabmatfournier) [reply kafka-connect/src/main/java/io/tabular/iceberg/connect/channel/Coordinator.java:100]: Sounds like a good reason for both.  The whole point of these logs are to help diagnose when a worker isn't working properly, as that is a lot of support cases in OSS.

File: kafka-connect/src/main/java/io/tabular/iceberg/connect/channel/Coordinator.java (hunk 1 above)
Comment (tabmatfournier) [reply kafka-connect/src/main/java/io/tabular/iceberg/connect/channel/Coordinator.java:132]: There is a flow where you don't get here and I wanted to capture it

**Pull request merged**